data/pools/transactionPool_test.go

```diff
@@ -988,3 +1116,96 @@ func BenchmarkTransactionPoolPending(b *testing.B) {
 		})
 	}
 }
+
+func BenchmarkTransactionPoolSteadyState(b *testing.B) {
+	poolSize := 100000
+
+	fmt.Printf("BenchmarkTransactionPoolSteadyState: N=%d\n", b.N)
+
+	numOfAccounts := 100
+	// Genereate accounts
+	secrets := make([]*crypto.SignatureSecrets, numOfAccounts)
+	addresses := make([]basics.Address, numOfAccounts)
+
+	for i := 0; i < numOfAccounts; i++ {
+		secret := keypair()
+		addr := basics.Address(secret.SignatureVerifier)
+		secrets[i] = secret
+		addresses[i] = addr
+	}
+
+	l := makeMockLedger(b, initAccFixed(addresses, 1<<32))
+	transactionPool := MakeTransactionPool(l, exponentialGrowth, poolSize, false)
+
+	var signedTransactions []transactions.SignedTxn
+	for i := 0; i < b.N; i++ {
+		var receiver basics.Address
+		crypto.RandBytes(receiver[:])
+		tx := transactions.Transaction{
+			Type: protocol.PaymentTx,
+			Header: transactions.Header{
+				Sender:      addresses[i%numOfAccounts],
+				Fee:         basics.MicroAlgos{Raw: uint64(rand.Int()%10000) + proto.MinTxnFee},
+				FirstValid:  0,
+				LastValid:   basics.Round(proto.MaxTxnLife),
+				GenesisHash: l.GenesisHash(),
+			},
+			PaymentTxnFields: transactions.PaymentTxnFields{
+				Receiver: receiver,
+				Amount:   basics.MicroAlgos{Raw: proto.MinBalance},
+			},
+		}
+		tx.Note = make([]byte, 8, 8)
+		crypto.RandBytes(tx.Note)
+
+		signedTx, err := transactions.AssembleSignedTxn(tx, crypto.Signature{}, crypto.MultisigSig{})
+		require.NoError(b, err)
+		signedTransactions = append(signedTransactions, signedTx)
+	}
+
+	b.StopTimer()
+	b.ResetTimer()
+	b.StartTimer()
+
+	poolTxnQueue := signedTransactions
+	var ledgerTxnQueue []transactions.SignedTxn
+
+	for len(poolTxnQueue) > 0 || len(ledgerTxnQueue) > 0 {
+		// Fill up txpool
+		for len(poolTxnQueue) > 0 {
+			stx := poolTxnQueue[0]
+			err := transactionPool.Remember(stx)
+			if err == nil {
+				poolTxnQueue = poolTxnQueue[1:]
+				ledgerTxnQueue = append(ledgerTxnQueue, stx)
+				continue
+			}
+			if strings.Contains(err.Error(), "transaction pool is full") {
+				break
+			}
+			require.NoError(b, err)
+		}
+
+		// Commit a block
```

Comment:
it seems to benchmark a few things that the pool is doing together:
-remember transactions
-remove pending txes when a block gets committed 
-creating a proposed block.

Maybe that's okay for this use case, I'm not sure if more detailed benchmarks (splitting the 3 operations) would be more insightful